### ISSUE TITLE
[FEATURE REQUEST] Allow custom mapping of 'brightness_step_down'

### ISSUE THREAD

**Issue by mvjt** [xaviml/controllerx#151]
# Feature Request
Allow brightness_step_down and brightness_step_up to be overridden (sent by IKEA 1744/Symfonisk when using Z2M config legacy: false. I`m using the MQTT integration but the others should also be supported.

### Is your feature request related to a problem?
Yes

I cannot override double and triple click from the Symfonisk remote.

### Describe the solution / feature you'd like
Allow brightness_step_down and brightness_step_up to be overridden (sent by IKEA 1744/Symfonisk when using Z2M config legacy: false. I`m using the MQTT integration but the others should also be supported.

Allow brightness_step_down and brightness_step_up to be overridden (sent by IKEA 1744/Symfonisk when using Z2M config legacy: false when using "mapping" in the ControllerX config. I`m using the MQTT integration but the others should also be supported.

### Describe alternatives you've considered
None

**Comment by xaviml** [comment]
Hi @mvjt,

Not sure if I fully understood your request, but you can override any mapping with [custom mapping](https://xaviml.github.io/controllerx/others/custom-controllers). The default behaviour for double and triple click for this controller used in media players is to go back and forward in your tracklist, but you can change it like:
```yaml
  example_app:
    module: controllerx
    class: E1744MediaPlayerController
    controller: my_frienly_name
    integration:
      name: z2m
      listen_to: mqtt
    media_player: media_player.my_media_player
    merge_mapping: # This keeps the default mapping and override the specified actions
       brightness_step_up: < Any action you want to use or call a service >
       brightness_step_down: < Any action you want to use or call a service >
```

If this is not what you were looking for, could you expand your idea a little bit more and also share your current configuration?

Cheers,
Xavi M.

**Comment by mvjt** [comment]
> Hi @mvjt,
> 
> Not sure if I fully understood your request, but you can override any mapping with [custom mapping](https://xaviml.github.io/controllerx/others/custom-controllers). The default behaviour for double and triple click for this controller used in media players is to go back and forward in your tracklist, but you can change it like:
> 
> ```yaml
>   example_app:
>     module: controllerx
>     class: E1744MediaPlayerController
>     controller: my_frienly_name
>     integration:
>       name: z2m
>       listen_to: mqtt
>     media_player: media_player.my_media_player
>     merge_mapping: # This keeps the default mapping and override the specified actions
>        brightness_step_up: < Any action you want to use or call a service >
>        brightness_step_down: < Any action you want to use or call a service >
> ```
> 
> If this is not what you were looking for, could you expand your idea a little bit more and also share your current configuration?
> 
> Cheers,
> Xavi M.

Thanks Xavi,

I will test this. I was using the following based on what I figured out from the docs.
`kok_media:
  module: controllerx
  class: E1744MediaPlayerController
  controller: zigbee2mqtt/kok_volume_control/action
  integration: mqtt
  media_player: media_player.kok
  mapping:
    click_brightness_down:
      - service: media_player/play_media
        data:
          entity_id: media_player.kok
          media_content_id: "http://sverigesradio.se/topsy/direkt/212-hi.mp3"
          media_content_type: "music"`

**Comment by xaviml** [comment]
Hi @mvjt,

You just need to change `click_brightness_down` by `brightness_step_down` and your configuration should work as expected.

Cheers,
Xavi M.

**Comment by mvjt** [comment]
I tried that first but it did not override. Let me try again...

**Comment by mvjt** [comment]
`kok_media:
  module: controllerx
  class: E1744MediaPlayerController
  controller: kok_volume_control
  integration:
    name: z2m
    listen_to: mqtt
  media_player: media_player.kok`

That does not even ork...Very weird...

**Comment by mvjt** [comment]
kok_media:
  module: controllerx
  class: E1744MediaPlayerController
  controller: kok_volume_control
  integration:
    name: z2m
    listen_to: mqtt
  media_player: media_player.kok
  merge_mapping:
    brightness_step_down:
      - service: media_player/play_media
        data:
          entity_id: media_player.kok
          media_content_id: "http://sverigesradio.se/topsy/direkt/212-hi.mp3"
          media_content_type: "music"


This does not work

**Comment by xaviml** [comment]
Could you wrap the yaml code like:
\`\`\`yaml
yaml code here
\`\`\`

The configuration I was referring is the following:

```yaml
kok_media:
  module: controllerx
  class: E1744MediaPlayerController
  controller: zigbee2mqtt/kok_volume_control/action
  integration: mqtt
  media_player: media_player.kok
  mapping:
    brightness_step_down:
      - service: media_player/play_media
        data:
          entity_id: media_player.kok
          media_content_id: "http://sverigesradio.se/topsy/direkt/212-hi.mp3"
          media_content_type: "music"
```

You can also do the following, which is the same:

```yaml
kok_media:
  module: controllerx
  class: E1744MediaPlayerController
  controller: kok_volume_control
  integration: 
    name: z2m
    listen_to: mqtt
  media_player: media_player.kok
  mapping:
    brightness_step_down:
      - service: media_player/play_media
        data:
          entity_id: media_player.kok
          media_content_id: "http://sverigesradio.se/topsy/direkt/212-hi.mp3"
          media_content_type: "music"
```

Could you share your `appdaemon.yaml`? Did you add the MQTT information in there like explained in [here](https://xaviml.github.io/controllerx/others/integrations#mqtt)?

Cheers,
Xavi M

**Comment by mvjt** [comment]
Yes I wrapped it in previous post. The control works if I use MQTT only but not when using the z2m + listen... 

```
    MQTT:
      type: mqtt
      namespace: mqtt # This is important
      client_host: 192.168.1.241
      #client_user: XXXXX
      #client_password: XXXXX
```

**Comment by xaviml** [comment]
Are you using the latest ControllerX version? (v4.0.0)

Could you share the AppDaemon logs when the controller does not work?

**Comment by mvjt** [comment]
I downloaded the new version yesterday but apparently it shows up as 3.30...I need to see what I did wrong when unzipping the archive.Thanks

**Comment by xaviml** [comment]
When updating ControllerX, it is necessary to restart the AppDaemon server/addon.

**Comment by mvjt** [comment]
That was it! Cheers. Thanks for quick support...

**Comment by xaviml** [comment]
The custom mapping is now working then? Perfect!

No worries, thank you for your patience and for using this project :)

Cheers,
Xavi M.

**Comment by mvjt** [comment]
Yes the custom mapping works too!

**Comment by xaviml** [comment]
De nada :)
